src/main/java/com/gym/gymsystem/controller/TrainingController.java

```diff
@@ -0,0 +1,151 @@
+package com.gym.gymsystem.controller;
+
+import com.gym.gymsystem.dto.trainer.TrainerInfo;
+import com.gym.gymsystem.dto.training.AddTrainingRequest;
+import com.gym.gymsystem.dto.training.TrainingInfo;
+import com.gym.gymsystem.dto.training.TrainingInfoForTrainer;
+import com.gym.gymsystem.dto.training.UpdateTraineeTrainersRequest;
+import com.gym.gymsystem.dto.user.Message;
+import com.gym.gymsystem.entity.Trainer;
+import com.gym.gymsystem.entity.Training;
+import com.gym.gymsystem.entity.TrainingType;
+import com.gym.gymsystem.exception.CustomAccessDeniedException;
+import com.gym.gymsystem.service.AuthorizationService;
+import com.gym.gymsystem.service.TraineeService;
+import com.gym.gymsystem.service.TrainerService;
+import com.gym.gymsystem.service.TrainingService;
+import jakarta.validation.Valid;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/api/training")
+public class TrainingController {
+    private final TrainingService trainingService;
+    private final TraineeService traineeService;
+    private final TrainerService trainerService;
+    private final AuthorizationService authorizationService;
+
+    public TrainingController(TrainingService trainingService, TraineeService traineeService, TrainerService trainerService, AuthorizationService authorizationService) {
+        this.trainingService = trainingService;
+        this.traineeService = traineeService;
+        this.trainerService = trainerService;
+        this.authorizationService = authorizationService;
+    }
+
+    @GetMapping("/trainers/not-assigned")
+    public ResponseEntity<List<TrainerInfo>> getNotAssignedActiveTrainers(
+            @RequestParam("findUsername") String findUsername) {
+        if (!authorizationService.isAdmin() && !authorizationService.isTrainer() && !authorizationService.isAuthenticatedUser(findUsername)) {
+            throw new CustomAccessDeniedException("You do not have permission to get this trainee's not assigned trainers.");
+        }
+        List<TrainerInfo> trainerInfos = trainingService.getTrainersNotAssignedToTrainee(findUsername);
+        return ResponseEntity.ok(trainerInfos);
+    }
+
+    @PutMapping("/trainers")
+    public ResponseEntity<List<TrainerInfo>> updateTraineeTrainers(
+            @RequestParam("findUsername") String findUsername,
```

Comment:
I guess this is trainer username. if so, it better to use path param for this and named like trainerUsername